openassessment/xblock/ui_mixins/mfe/test_page_context_serializer.py

```diff
@@ -606,6 +619,29 @@ def test_self_assessment_not_available(self, xblock):
 
         self.assertNestedDictEquals(expected_data, progress_data)
 
+    @scenario('data/grade_scenario_self_staff.xml', user_id='Alan')
+    def test_self_staff_assessment__waiting(self, xblock):
+        # Given I am on the staff step and waiting for submissions
+        self.create_submission_and_assessments(xblock, 'submission_text', [], [], SELF_ASSESSMENT)
+        xblock.workflow_data.workflow['status'] = 'waiting'
+
+        # When I ask for progress
+        progress_data = ProgressSerializer(xblock).data
+
+        # Expect active step to be staff instead of waiting
+        self.assertEqual('staff', progress_data['activeStepName'])
+
+    @scenario('data/grade_scenario_self_staff_not_required.xml', user_id='Alan')
+    def test_self_staff_assessment__not_required(self, xblock):
+        # Given I am on the staff step and waiting for submissions
+        self.create_submission_and_assessments(xblock, 'submission_text', [], [], SELF_ASSESSMENT)
+
+        # When I ask for progress
+        progress_data = ProgressSerializer(xblock).data
+
+        # Expect active step to be staff instead of waiting
```

Comment:
This comment doesn't reflect what's happening. I think this test is actually not necessary because as soon as we submit the self assessment in this case, since the staff is not required (and therefore auto-generated) it's just skipped and we go straight to done rather than waiting